numpy/core/arrayprint.py

```diff
@@ -1339,13 +1339,29 @@ def _format_non_nat(self, x):
 
 
 class SubArrayFormat:
-    def __init__(self, format_function):
+    def __init__(self, format_function, **options):
         self.format_function = format_function
+        self.threshold = options['threshold']
+        self.edge_items = options['edgeitems']
+
+    def __call__(self, a):
+        self.summary_insert = "..." if a.size > self.threshold else ""
+        return self.format_array(a)
+
+    def format_array(self, a):
+        if np.ndim(a) == 0:
+            return self.format_function(a)
```

Comment:
I think such a dtype isn't really possible, but lets keep it.

Reply:
This line was necessary, and I couldn't rely on `a.ndim` either... I forget exactly which tests failed, though, just remember that I was surprised too!